bag.go

```diff
@@ -8,10 +8,93 @@ import (
 	"os"
 	"path/filepath"
 	"regexp"
+	"strings"
 )
 
-var manifestPtn = regexp.MustCompile("manifest-.*\\.txt$")
-var tagmanifestPtn = regexp.MustCompile("tagmanifest-.*\\.txt$")
+type Bag struct {
+	Path         string
+	Name         string
+	AbsPath      string
+	Payload      Payload
+	BagInfo      TagSet
+	BagIt        TagSet
+	Manifests    ManifestRefs
+	TagManifests ManifestRefs
+}
+
+func GetExistingBag(path string) (Bag, error) {
+	bag := Bag{}
+
+	//check directory exists
+	if err := directoryExists(path); err != nil {
+		return bag, err
+	}
+
+	//set path
+	bag.Path = path
+
+	//set absolute path
+	var err error
+	bag.AbsPath, err = getABS(bag.Path)
+	if err != nil {
+		return bag, err
+	}
+
+	//set name
+	pathSplit := strings.Split(bag.AbsPath, string(os.PathSeparator))
+	bag.Name = pathSplit[len(pathSplit)-1]
+
+	//get Bagit.txt
+	bag.BagIt, err = NewTagSet("bagit.txt", bag.Path)
+	if err != nil {
+		return bag, err
+	}
+
+	//get Baginfo.txt
+	bag.BagInfo, err = NewTagSet("bag-info.txt", bag.Path)
+	if err != nil {
+		return bag, err
+	}
+
+	//get Manifests
+	bag.Manifests, err = GetManifests(bag.Path)
+	if err != nil {
+		return bag, err
+	}
+
+	//get Tag Manifests
+	bag.TagManifests, err = GetTagManifests(bag.Path)
+	if err != nil {
+		return bag, err
+	}
+
+	bag.Payload = make(map[string]os.FileInfo)
+	if err := loadPayload(&bag); err != nil {
+		return bag, err
+	}
+
+	return bag, nil
+}
+
+func (b Bag) ListPayloadFiles() {
+	for _, fi := range b.Payload {
+		fmt.Println(fi.Name())
+	}
+}
+
+func (b Bag) GetAbsolutePath() (string, error) {
+	abs, err := filepath.Abs(b.Path)
```

Comment:
Could you just return `b.AbsPath` here?  If not, would there be an advantage to storing the absolute path in the bag struct? (if so, you would need to change the function signature to pass `b *Bag`)

Reply:
yes, i think this is confusion from iteration